datalab/datalab_session/data_operations/input_data_handler.py

```diff
@@ -27,15 +29,27 @@ def __init__(self, basename: str, source: str = None) -> None:
     self.source = source
     self.exit_stack = ExitStack()
     self.fits_file = self.exit_stack.enter_context(get_fits(basename, source))
-    self.sci_data = get_hdu(self.fits_file, 'SCI').data
+    self.sci_hdu = get_hdu(self.fits_file, 'SCI')
```

Comment:
We were already getting the SCI HDU and copying it out of the file, so this just saves the reference to it in this class so we can pull out the header if we need it without opening the file and copying the HDU again.